### PR TITLE
User interface

### PR DESCRIPTION
**Pull request by ikraamg**
Dear TSE,

_In this step we created a basic game flow that takes input from the player and prompts a message with the board layout. We are not diving into game details yet._

1. Required input from the player maintained by the **question** method.
2. The game board which basically is a 3 by 3 grid is displayed by the **show_board** method.
3. **puts/print**, **get.chomp** methods used to interact with the player.
4. Able to run with the executable file.
5. Basic iteration is fulfilled to mock the playing with method **play**.

[Demo Link Milestone-2](https://repl.it/@kblycaglayan/TTTmilestone2)

With all of our respects...

### CODE REVIEW

**Review by whiz25 — CHANGES_REQUESTED** [review]
## Good Progress :+1: 
Your code looks good so far but we need to fix a few things before we can merge and proceed.
Check my comments for suggestions.

### Martin Afani

File: bin/main.rb [hunk 1]
```diff
@@ -1,3 +1,65 @@
...
+  draw_grid(board_show)
+end
+
+def draw_grid(array)
+  hypens = '-------------------'
+  hypens_w_nl = "-------------------\n"
+  puts "\n  " + hypens
+  3.times do |i|
+    print '  | ', array[i * 3], ' | ', array[i * 3 + 1], ' | ', array[i * 3 + 2], " |\n  #{hypens_w_nl if i < 2}"
+  end
```
Comment (whiz25) [inline bin/main.rb:26]: - [x]  use a terminating variable
For example
```ruby
game_on = true
while game_on
# include logic for game here
if winner
  game_on = false
else
# include logic for draw
end
end

File: bin/main.rb [hunk 2]
```diff
@@ -1,3 +1,65 @@
...
+
+  board = create_board
+
+  show_board(board)
+
+  9.times do |turn|
+    print "\nThis is turn: #{turn + 1} out of 9\n "
+    question(turn)
+    puts '(Your move is displayed on the board below)'
+    show_board(board)
```
Comment (whiz25) [inline bin/main.rb]: - [x] use a terminating variable here too

**Comment by kubilaycaglayan** [comment]
Dear TSE,

Thank you very much for your feedback. We appreciate a lot and see this as a big learning opportunity. We did the required changes according to your comments. 

Please note that in your first comment there is a drawing that is meaningless to add termination variable. Please let us know if we can do anything more for you...

**following block added:**
  ```
game_on = true
  turn = 0
  winner = false
  while game_on
    print "\nThis is turn: #{turn + 1} out of 9\n "
    question(turn)
    puts '(Your move is displayed on the board below)'
    show_board(board)
    turn += 1
    winner = true if turn == 8
    game_on = false if winner
  end
```
[LIVE DEMO](https://repl.it/@kblycaglayan/TTTmilestone2)

With all of our respects...

**Review by kobiyoyo — CHANGES_REQUESTED** [review]
# Project Approved :thumbsup:
## Hello Team ,  You have done a great job with this project. :clap:

**Pull request merged**